### ISSUE TITLE
Social Icons Block: Updated to Labelling and Focus for Better Accessibility

### ISSUE THREAD

**Issue by donalirl** [WordPress/gutenberg#22412]
**Describe the bug**
I am reporting this on behalf of a visually-impaired user who reports that the Social Icons block needs better labelling and also a correct focus to be usable with Apple Voiceover.

**To reproduce**
1. Go to a new page or post and activate Apple Voiceover (screen reader)
2. Add a Social Icons block and attempt to add icons to the block using the screen reader only.

**Actual behavior**
Apple Voiceover calls out the Social Icons block as just "button", so it is not clear to a blind person what is selected.

They tap on the button so they can paste the URL but the Voiceover cursor does not focus on the URL field, so the have to explore the screen using their finger to find the URL field. They can double-tap to open it but sometimes cannot find the URL box at all. 

**Expected behavior**
They expected the block to be labeled to say something like "social links" so that screen reader users know what that they are using.

For the cursor, what needs to happen is that once they tap on the URL field, the Voiceover cursor should then be focused in the URL text box, and Voiceover would indicate that by saying something like "URL text box".

**Editor version (please complete the following information):**
WordPress version: 5.4
Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? Default

**Smartphone (please complete the following information):**
Device: iPod Touch
Browser: Safari Latest
Screen reader: Apple Voiceover

**Comment by annezazu** [comment]
Noting that this came up in a session today with a blind WordPress developer who noted the same problem using the latest version of Gutenberg (10.1) in the Full Site Editing experience.

**Comment by carolinan** [comment]
I am not able to reproduce this in Safari and voiceover **on mac.**
Voiceover presents the block as "Social icons, block. Document". 
If I navigate to the inner add block button, it announces "Add block, menu popup collapsed, button, Social icons".
If I select an existing inner icon block it announces "(name),button, Social icons, block. Document"

I did **not** test with a touch device.

**Comment by hypest** [comment]
Looks like the issue is encountered in a web browser so, I will remove the `Mobile App - i.e. Android or iOS` label.

**Comment by ndiego** [comment]
I just tested Gutenberg 13.7.3 on a iPhone using VoiceOver and I am not able to replicate the issue. I am getting similar results to @carolinan. The Social Icon is clearly announced and I am able to navigate to the URL field and enter a link. I am going to close this issue, but if anyone is able to replicate in the future, feel free to reopen.
